h2o-py/h2o/model/model_base.py

```diff
@@ -459,6 +459,8 @@ def summary(self):
         model = self._model_json["output"]
         if "model_summary" in model and model["model_summary"] is not None:
             return model["model_summary"]
+        if "glm_model_summary" in model and model["glm_model_summary"] is not None:
+            return model["glm_model_summary"]
```

Comment:
really? even for such a simple property, GLM is a special case?

Reply:
Not GLM, GAM is a special case.

Reply:
ok... I'm still not a fan of those algo-specific logic in abstract classes: couldn't we add sth like
```
def _get_model_summary(self):
    …
```
and override it for GAM?

Otherwise, the logic for a given algo is scattered among multiple files, making it easy to miss something (see `estimator_base.py` for many nasty examples like this, that I'd like to change one day).

Btw, a slightly more elegant syntax:
```
model.get('glm_model_summary')
```
```

Reply:
@sebhrusen Since there is a similar thing for scoring history in GAM, I would place the following to the bindings generation of GAM as `class_extension`.

```python
    def summary(self):
        """Print a detailed summary of the model."""
        model = self._model_json["output"]
        if "glm_model_summary" in model and model["glm_model_summary"] is not None:
            return model["glm_model_summary"]
        print("No model summary for this model")

    def scoring_history(self):
        """
        Retrieve Model Score History.

        :returns: The score history as an H2OTwoDimTable or a Pandas DataFrame.
        """
        model = self._model_json["output"]
        if "glm_scoring_history" in model and model["glm_scoring_history"] is not None:
            return model["glm_scoring_history"].as_data_frame()
        print("No score history for this model")
```

This way I keep it only in GAM related code and if GAM decides to use conventional names it would involve only changing the code related to GAM (and we wouldn't have a useless (after unifying the interface) method `_get_model_summary()` which would be basically the same thing as the `summary()` except for the `print` statement/function). (maybe I am too optimistic about unifying the interface)

WDYT?

Reply:
yes, I think it's the best approach, the most important for me is to see clearly all the GAM changes in one place.

Reply:
Which reminds me that a while ago, I wanted  to create an annotation to automatically inherit  `doc`  from parent.
It's a nice Friday task.